src/libraries/System.Reflection.Emit/tests/ModuleBuilder/ModuleBuilderDefineInitializedData.cs

```diff
@@ -59,5 +59,50 @@ public void DefineInitializedData_CreateGlobalFunctionsCalled_ThrowsInvalidOpera
             module.CreateGlobalFunctions();
             Assert.Throws<InvalidOperationException>(() => module.DefineInitializedData("MyField2", new byte[] { 1, 0, 1 }, FieldAttributes.Public));
         }
+
+        [Fact]
+        public void DefineInitializedData_EnsureAlignmentIsMinimumNeededForUseOfCreateSpan()
+        {
+            ModuleBuilder module = Helpers.DynamicModule();
+
+            // Create static field data in a variety of orders that requires the runtime to actively apply alignment
+            // RuntimeHelpers.CreateSpan requires data to be naturally aligned within the "PE" file. At this time CreateSpan only
+            // requires alignments up to 8 bytes.
+            FieldBuilder field1Byte = module.DefineInitializedData("Field1Byte", new byte[] { 1 }, FieldAttributes.Public);
```

Comment:
Let's also have a test for 12 bytes, which, I believe, should result in 4-byte alignment.

Reply:
It will result in 4 byte alignment on CoreCLR, but possibly only 2 byte on Mono. I think though, that the current set of tests is adequate.